modules/standard/fun/ding_controller.py

```diff
@@ -0,0 +1,137 @@
+import random
+
+from core.command_param_types import Any
+from core.decorators import command, instance
+from core.registry import Registry
+
+
+@instance()
+class DingController:
+    """
+        Port of Budabot fun module
+    """
+
+    def inject(self, registry: Registry):
+        self.command_alias_service = registry.get_instance(
+            "command_alias_service")
+
+    def start(self):
+        self.command_alias_service.add_alias("levelup", "ding")
+
+    @command(command="ding", params=[Any("level", is_optional=True)], access_level="all",
+             description="Show a random ding message, congratz!")
+    def ding_command(self, request, level: str):
+        if level is None:
+            text_options = [
+                "Yeah yeah gratz, I would give you a better response but you didn't say what level you dinged.",
+                "Hmmm, I really want to know what level you dinged, but gratz anyways nub.",
+                "When are you people going to start using me right! Gratz for your level though.",
+                "Gratz! But what are we looking at? I need a level next time."]
+            return random.choice(text_options)
+        else:
+            try:
+                current_level = int(float(level))
```

Comment:
If you do Int("level"...), you'll be able to omit the try-catch thing you got going, as level will be an in (remove the explicit type in the argument list, though)